src/utilities/byProgram.js

```diff
@@ -23,7 +23,14 @@ import { SETTINGS_KEYS } from '../settings';
  */
 const getAllPrograms = (settings, database) => {
   const thisStoresNameID = settings.get(SETTINGS_KEYS.THIS_STORE_NAME_ID);
-  const thisStoresTags = settings.get(SETTINGS_KEYS.THIS_STORE_TAGS);
+  const thisStoresTags = database
+    .objects('NameTag')
+    .filtered(
+      'subquery(nameTagJoins, $joins, $joins.name.id == $0).@count > 0',
+      settings.get(SETTINGS_KEYS.THIS_STORE_NAME_ID)
+    )
+    .map(({ description }) => description.toLowerCase());
```

Comment:
Not pretty - hoping to get the functionality required done, then refactor a bit more.